airbyte-integrations/connectors/source-relational-db/src/test/java/io/airbyte/integrations/source/relationaldb/AbstractDbSourceTest.java

```diff
@@ -0,0 +1,125 @@
+package io.airbyte.integrations.source.relationaldb;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.commons.functional.CheckedConsumer;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.commons.resources.MoreResources;
+import io.airbyte.commons.util.AutoCloseableIterator;
+import io.airbyte.db.AbstractDatabase;
+import io.airbyte.protocol.models.AirbyteStateMessage;
+import io.airbyte.protocol.models.AirbyteStateMessage.AirbyteStateType;
+import io.airbyte.protocol.models.CommonField;
+import io.airbyte.protocol.models.JsonSchemaType;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test suite for the {@link AbstractDbSource} class.
+ */
+public class AbstractDbSourceTest {
+
+  @Test
+  void testDeserializationOfState() throws IOException {
+    final AbstractDbSource dbSource = new TestAbstractDbSource();
+    final JsonNode config = mock(JsonNode.class);
+
+    final String globalStateJson = MoreResources.readResource("states/global.json");
+    final String legacyStateJson = MoreResources.readResource("states/legacy.json");
+    final String streamStateJson = MoreResources.readResource("states/per_stream.json");
+
+    final JsonNode globalState = Jsons.deserialize(globalStateJson);
+    final JsonNode legacyState = Jsons.deserialize(legacyStateJson);
+    final JsonNode streamState = Jsons.deserialize(streamStateJson);
+
+    // Legacy state
```

Comment:
Should it be 4 different tests? (one for legacy, one for global...)

Reply:
I can break it out if that's easier to read/reason about.

Reply:
Done.